Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2393,7 +2391,23 @@ static String preferredFilenameForElement(const HTMLImageElement& element)
     return nil;
 }
 
-static RetainPtr<NSAttributedString> attributedStringWithAttachmentForElement(HTMLImageElement& element)
+static RetainPtr<NSFileWrapper> fileWrapperForElement(const HTMLAttachmentElement& element)
+{
+    auto identifier = element.uniqueIdentifier();
+
+    RetainPtr data = [(NSString *)identifier dataUsingEncoding:NSUTF8StringEncoding];
+    if (!data)
+        return nil;
+
+    // Use a filename prefixed with a sentinal value to indicate that the data is corresponding
+    // to an existing HTMLAttachmentElement.
+
+    RetainPtr wrapper = adoptNS([[NSFileWrapper alloc] initRegularFileWithContents:data.get()]);
+    [wrapper setPreferredFilename:makeString(WebContentReader::placeholderAttachmentFilenamePrefix, identifier)];
```

Comment:
This is...pretty hacky, but I'm not sure I have a better suggestion immediately. I suppose it's okay for now, as long as a maliciously-crafted email can't cause, say, a crash in either the web process or Mail by purposely leaving an attachment named `webkit-attachment-XXXXX`.

Does it work if you make this a separate (custom) key in the `NSAttributedString`, that maps to the attachment ID?

Reply:
> Does it work if you make this a separate (custom) key in the NSAttributedString, that maps to the attachment

It does not, since NSHtMLWriter just ends up dropping it